staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -223,7 +223,14 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	}
 
 	// WithContext creates a shallow clone of the request with the new context.
-	newReq := req.WithContext(context.Background())
+	ctx := context.Background()
+	// if the original request has a deadline, we should honor that deadline for our proxied request
+	if deadline, ok := req.Context().Deadline(); ok {
+		var cancelFn context.CancelFunc
+		ctx, cancelFn = context.WithDeadline(ctx, deadline)
```

Comment:
It seems like this breaks the property where if the enclosing request is canceled, the child request is also canceled.

Reply:
That's pre-existing, we all noticed that, but none know why.  I opened an issue to fix in 1.17